src/DssSpell.sol

```diff
@@ -55,172 +36,62 @@ contract DssSpellAction is DssAction {
-        DssExecLib.increaseGlobalDebtCeiling(1_000);
-        // Set the ilk debt ceiling
-        DssExecLib.setIlkDebtCeiling(ilk, 1_000);
-
-        // No dust
-        // DssExecLib.setIlkMinVaultAmount(ilk, 0);
-
-        // 3% stability fee
-        DssExecLib.setIlkStabilityFee(ilk, THREE_PCT_RATE, false);
-
-        // collateralization ratio 100%
-        DssExecLib.setIlkLiquidationRatio(ilk, 10_000);
-
-        // poke the spotter to pull in a price
-        DssExecLib.updateCollateralPrice(ilk);
-
-        // give the urn permissions on the join adapter
-        DssExecLib.authorize(MCD_JOIN_RWA001_A, RWA001_A_URN);
-
-        // set up the urn
-        Hopeable(RWA001_A_URN).hope(RWA001_OPERATOR);
-
-        // set up output conduit
-        Hopeable(RWA001_A_OUTPUT_CONDUIT).hope(RWA001_OPERATOR);
-
-        // Authorize the SC Domain team deployer address on the output conduit during introductory phase.
-        //  This allows the SC team to assist in the testing of a complete circuit.
-        //  Once a broker dealer arrangement is established the deployer address should be `deny`ed on the conduit.
-        Kissable(RWA001_A_OUTPUT_CONDUIT).kiss(SC_DOMAIN_DEPLOYER_07);
-
-        // add RWA-001 contract to the changelog
-        DssExecLib.setChangelogAddress("RWA001", RWA001_GEM);
-        DssExecLib.setChangelogAddress("PIP_RWA001", pip);
-        DssExecLib.setChangelogAddress("MCD_JOIN_RWA001_A", MCD_JOIN_RWA001_A);
-        DssExecLib.setChangelogAddress("MIP21_LIQUIDATION_ORACLE", MIP21_LIQUIDATION_ORACLE);
-        DssExecLib.setChangelogAddress("RWA001_A_URN", RWA001_A_URN);
-        DssExecLib.setChangelogAddress("RWA001_A_INPUT_CONDUIT", RWA001_A_INPUT_CONDUIT);
-        DssExecLib.setChangelogAddress("RWA001_A_OUTPUT_CONDUIT", RWA001_A_OUTPUT_CONDUIT);
+        // Rates Proposal
+        DssExecLib.setIlkStabilityFee("UNIV2DAIETH-A", THREE_PCT, true);
+        DssExecLib.setIlkStabilityFee("UNIV2USDCETH-A", THREE_PT_FIVE_PCT, true);
+        DssExecLib.setIlkStabilityFee("UNIV2WBTCETH-A", FOUR_PT_FIVE_PCT, true);
+        DssExecLib.setIlkStabilityFee("UNIV2ETHUSDT-A", FIVE_PCT, true);
+        DssExecLib.setIlkStabilityFee("UNIV2LINKETH-A", FIVE_PCT, true);
+        DssExecLib.setIlkStabilityFee("UNIV2UNIETH-A", FIVE_PCT, true);
+        DssExecLib.setIlkStabilityFee("UNIV2AAVEETH-A", FIVE_PCT, true);
+        DssExecLib.setIlkStabilityFee("MANA-A", FOUR_PCT, true);
+        DssExecLib.setIlkStabilityFee("UNIV2WBTCDAI-A", TWO_PCT, true);
+        DssExecLib.setIlkStabilityFee("RENBTC-A", FIVE_PCT, true);
+
+        // Onboarding ETH-C
+        CollateralOpts memory ETH_C = CollateralOpts({
+            ilk: "ETH-C",
+            gem: DssExecLib.getChangelogAddress("ETH"),
+            join: MCD_JOIN_ETH_C,
+            flip: MCD_FLIP_ETH_C,
+            pip: DssExecLib.getChangelogAddress("PIP_ETH"),
+            isLiquidatable: true,
+            isOSM: true,
```

Comment:
So this adds a bunch of duplicate authorizations and whitelisting, which we don't need; however, we do need this:
https://github.com/makerdao/dss-exec-lib/blob/012e177e0dc592b940d6bfe836df11e780ba6499/src/DssExecLib.sol#L727-L734

Is this everyone's assessment here too; that we need `isOSM`?

Reply:
This is the trade off by using DssExecLib. There are a number of places that are not so gas efficient, but I think it's worthwhile for safety we get.